### PR TITLE
Update to tokio 0.3

### PR DESCRIPTION
**Pull request by sfackler**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -194,19 +193,18 @@ impl<S> AsyncRead for SslStream<S>
...
-
     fn poll_read(
         mut self: Pin<&mut Self>,
         ctx: &mut Context<'_>,
-        buf: &mut [u8],
-    ) -> Poll<io::Result<usize>> {
-        self.with_context(ctx, |s| cvt(s.read(buf)))
+        buf: &mut ReadBuf<'_>,
+    ) -> Poll<io::Result<()>> {
+        self.with_context(ctx, |s| match cvt(s.read(buf.initialize_unfilled()))? {
```
Comment (sfackler) [inline src/lib.rs:201]: It's unfortunate that we're initializing here - I think the best route forward before AsyncRead is in std or there's a tokio 1.0 is probably to add a read method to rust-openssl which operates directly on `&mut {MaybeUninit<u8>]`.

**Comment by alexcrichton** [comment]
I'm so disconnected from the async ecosystem nowadays it probably doesn't make sense for this crate to live under me any more, if you'd like should I transfer it to you? (the repo, that is)

**Comment by sfackler** [comment]
Sure, works for me.

**Comment by alexcrichton** [comment]
Oh I think you'll need to delete your own fork, even renaming it I think doesn't work.

**Comment by sfackler** [comment]
Done! I'm surprised this PR didn't get auto-closed.

**Comment by alexcrichton** [comment]
👍

**Pull request merged**